### ISSUE TITLE
Build failure: ceph (on `staging`)

### ISSUE THREAD

**Issue by emilazy** [NixOS/nixpkgs#359723]
Ceph pins an old `cryptography` for arcane reasons; the rest of the ecosystem has moved beyond it which is causing this failure on `staging` after the latest batch of Python updates:

```
┃             ┌─ ⚠ python3.11-trustme-1.2.0 failed with exit code 1 after ⏱ 9s in pytestCheckPhase
┃          ┌─ ⏸ python3.11-requests-toolbelt-1.0.0
┃       ┌─ ⏸ python3.11-cheroot-10.0.1
┃    ┌─ ⏸ python3.11-cherrypy-18.10.0
┃ ┌─ ⏸ python3-3.11.10-env
┃ ⏸ ceph-19.2.0
```

A `trustme` test fails because it is calling `cryptography` methods that are not present in the version Ceph pins. It looks like it’s a [convenience library for writing tests](https://github.com/python-trio/trustme), so it’s possible that the fix will be as simple as turning off tests in `requests-toolbelt`.

cc @nh2

**Comment by nh2** [comment]
Linking related previous PR here that introduces the pinning of `cryptography`:

* https://github.com/NixOS/nixpkgs/pull/334286

**Comment by emilazy** [comment]
Going to maybe look into this tomorrow, but @nh2 do you think it might be a good idea to ping https://tracker.ceph.com/issues/64213 again as it’s been a few months?

**Comment by shymega** [comment]
This is present in 24.11 - and unstable, which seems to hit `cryptography` as described in the OP.

**Comment by jfly** [comment]
I took a stab at this, and quickly got out of my depth. I put together a draft PR here: https://github.com/NixOS/nixpkgs/pull/369622. Perhaps it's a useful starting point for someone.

**Comment by shymega** [comment]
> This is present in 24.11 - and unstable, which seems to hit `cryptography` as described in the OP.

Interestingly, it doesn't seem present in 24.11 now - can others confirm? In unstable, I'm still hitting the `cryptography` build error.

**Comment by jfly** [comment]
> Interestingly, it doesn't seem present in 24.11 now - can others confirm?

Yeah, I'm able to substitute ceph directly from cache.nixos.org when I run `nix build nixpkgs/nixos-24.11#ceph`.

<hr>

For the record, once we get past the Python issue described here (which will be addressed by https://github.com/NixOS/nixpkgs/pull/369622), we're going to hit a C++ compilation issue. I've filed https://github.com/NixOS/nixpkgs/issues/369777 to track that.

**Comment by jfly** [comment]
@jkotran, I ran into this in the exact same way as you ;)

One workaround is to disable ceph. From https://github.com/jfly/snow/blob/b72ef014fede1634b5ba0da04685ff1f845e7b29/hosts/kent/printer.nix#L58:

```nix
pkgs.sambaFull.override {
  # Workaround for https://github.com/NixOS/nixpkgs/issues/359723
  enableCephFS = false;
};
```

**Comment by Nairou** [comment]
> ```
> ┃             ┌─ ⚠ python3.11-trustme-1.2.0 failed with exit code 1 after ⏱ 9s in pytestCheckPhase
> ┃          ┌─ ⏸ python3.11-requests-toolbelt-1.0.0
> ┃       ┌─ ⏸ python3.11-cheroot-10.0.1
> ┃    ┌─ ⏸ python3.11-cherrypy-18.10.0
> ┃ ┌─ ⏸ python3-3.11.10-env
> ┃ ⏸ ceph-19.2.0
> ```

Apologies for the noise, but what tool generates this nested dependency list?

**Comment by SamLukeYes** [comment]
> what tool generates this nested dependency list?

I think it is [nix-output-monitor](https://code.maralorn.de/maralorn/nix-output-monitor)

**Comment by jkotran** [comment]
> @jkotran, I ran into this in the exact same way as you ;)
> 
> One workaround is to disable ceph. From https://github.com/jfly/snow/blob/b72ef014fede1634b5ba0da04685ff1f845e7b29/hosts/kent/printer.nix#L58:
> 
> ```nix
> pkgs.sambaFull.override {
>   # Workaround for https://github.com/NixOS/nixpkgs/issues/359723
>   enableCephFS = false;
> };
> ```

Thank you, that's a dynamite tip!
